kinder/pkg/cluster/manager/actions/smoke-test.go

```diff
@@ -66,10 +66,7 @@ func SmokeTest(c *status.Cluster, wait time.Duration) error {
 		return err
 	}
 
-	// wait a few seconds for svc to be reachable
-	time.Sleep(3 * time.Second)
-
-	err = checkNodePort(c, nodePort)
+	err = waitForNodePort(c, cp1, 30*time.Second, nodePort)
```

Comment:
the old code iterated over all the nodes.

so this should be:
```
for _, n := range c.K8sNodes() {
	err = waitForNodePort(c, n, wait, nodePort)
	if err != nil {
		return err
	}
}
```

this makes `wait` is a bit odd in the context of `SmokeTest()` as a whole, because it is now used in a couple of "wait" locations, so it can wait for a maximum of wait * 2

Reply:
one wait of handling that is passing `wait / 2` to all wait* calls.
it's default value is 5 minutes:
https://github.com/kubernetes/kubeadm/blob/cf387fefe4fd92ae6587b6a60c59d8b621a72095/kinder/cmd/kinder/do/do.go#L104

Reply:
I just kept the timeout to 30s for each of those calls. From what I saw the NodePort svc is ready within a couple of seconds. So the shorter timeout will help to fail fast.
The cluster that you get with when following the getting started guide has 3 nodes.

Reply:
How difficult would it be to pass multiple `try` funcs to the `waitFor` so that this happens in parallel?

Reply:
waitFor already supports this, but is a bit difficult as this is the sequence ATM:
- create deployment
- wait for pod
- expose port
- wait for port

if we swap "expose port" / "wait for pod" the pod might not be ready yet and expose will fail.
so i'd say leave it the way it is.